pallets/xyk/src/lib.rs

```diff
@@ -546,6 +546,48 @@ impl<T: Trait> Module<T> {
         }
     }
 
+    pub fn calculate_sell_price_id(
+        sold_token_id: TokenId,
+        bought_token_id: TokenId,
+        sell_amount: Balance,
+    ) -> Result<Balance, DispatchError> {
+        let input_reserve_u256: U256 = Pools::get((sold_token_id, bought_token_id))
+            .saturated_into::<u128>()
+            .into();
+        let output_reserve_u256: U256 = Pools::get((bought_token_id, sold_token_id))
+            .saturated_into::<u128>()
+            .into();
+        let sell_amount_u256: U256 = sell_amount.saturated_into::<u128>().into();
+
+        let input_amount_with_fee: U256 = sell_amount_u256 * 997;
+        let numerator: U256 = input_amount_with_fee * output_reserve_u256;
+        let denominator: U256 = input_reserve_u256 * 1000 + input_amount_with_fee;
```

Comment:
Could calculations be abstracted from the method? Currently we have price calculation code duplication, which is a significant flaw.
We should have only two source-of-truth function calculating price- one for sell price, one for buy price. The rest of the function should not know how the price is calculated.